mailgun.go

```diff
@@ -266,19 +266,26 @@ type MailgunImpl struct {
 	apiBase            string
 	domain             string
 	apiKey             string
+	webhookSigningKey  string
 	client             *http.Client
 	baseURL            string
 	overrideHeaders    map[string]string
 	capturedCurlOutput string
 }
 
 // NewMailGun creates a new client instance.
-func NewMailgun(domain, apiKey string) *MailgunImpl {
+// If you need to verify whether a Webhook originates from Mailgun, the webhookSigningKey is required.
+func NewMailgun(domain, apiKey string, webhookSigningKey ...string) *MailgunImpl {
```

Comment:
What if we needed some other keys later?
Ideally, this should be an option, but they are slow(https://www.evanjones.ca/go-functional-options-slow.html)

So having some `(mg *MailgunImpl) SetWebhookSigningKey(string)` should be OK.